libfuse/start.go

```diff
@@ -37,74 +37,51 @@ func Start(mounter Mounter, options StartOptions, kbCtx libkbfs.Context) *libfs.
 		}
 	}
 
+	log.Debug("Initializing")
+	interruptFn := func() {}
+	config, err := libkbfs.Init(
+		kbCtx, options.KbfsParams, nil, func() { interruptFn() }, log)
+	if err != nil {
+		return libfs.InitError(err.Error())
+	}
+	defer libkbfs.Shutdown()
+
 	log.Debug("Mounting: %s", mounter.Dir())
 	c, err := mounter.Mount()
 	if err != nil {
 		return libfs.MountError(err.Error())
 	}
-	var doneChan <-chan struct{}
-	var onInterruptFn func()
-
-	if c != nil {
-		defer c.Close()
-
-		doneChan = c.Ready
-		onInterruptFn = func() {
-			select {
-			case <-c.Ready:
-				// Was mounted, so try to unmount if it was successful.
-				if c.MountError == nil {
-					err = mounter.Unmount()
-					if err != nil {
-						return
-					}
-				}
+	defer mounter.Unmount()
 
-			default:
-				// Was not mounted successfully yet, so do nothing. Note that the mount
-				// could still happen, but that's a rare enough edge case.
-			}
-			libkbfs.Shutdown()
+	done := make(chan struct{})
+	if c != nil { // c can be nil for NoopMounter
+		interruptFn = func() {
+			mounter.Unmount()
 		}
 	} else {
-		// When there's no mount, declare ourselves done when we get
-		// an interrupt.
-		ch := make(chan struct{}, 1)
-		doneChan = ch
-		onInterruptFn = func() {
-			select {
-			case ch <- struct{}{}:
-				libkbfs.Shutdown()
-			default:
-			}
+		interruptFn = func() {
+			close(done)
 		}
 	}
 
-	log.Debug("Initializing")
-
-	config, err := libkbfs.Init(kbCtx, options.KbfsParams, nil, onInterruptFn, log)
-	if err != nil {
-		return libfs.InitError(err.Error())
-	}
-
-	defer libkbfs.Shutdown()
-
 	if c != nil {
+		<-c.Ready
```

Comment:
`<-c.Ready` is closed when the mount is ready or has failed ([godoc](https://godoc.org/bazil.org/fuse#Conn)). So I thought it makes sense to do it before the Serve. I'll test on Linux to make sure before merging.